File: datasets/lua/HumanEval_131_digits.lua
-- Given a positive integer n, return the product of the odd digits.
-- Return 0 if all digits are even.
-- For example:
-- digits(1)  == 1
-- digits(4)  == 0
-- digits(235) == 15
local function digits(n)
  local product = 1
  local digits = tostring(n)
  for i = 1, #digits do
    local digit = tonumber(digits:sub(i, i))
    if digit % 2 == 1 then
      product = product * digit
    end
  end
  return product
end

lu = require('luaunit')

function test_humaneval()
local candidate = digits
    lu.assertEquals(candidate(5), 5)
    lu.assertEquals(candidate(54), 5)
    lu.assertEquals(candidate(120), 1)
    lu.assertEquals(candidate(5014), 5)
    lu.assertEquals(candidate(98765), 315)
    lu.assertEquals(candidate(5576543), 2625)
    lu.assertEquals(candidate(2468), 0)
end

os.exit(lu.LuaUnit.run())

File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_131_digits.pl
# Given a positive integer n, return the product of the odd digits.
# Return 0 if all digits are even.
# For example:
# digits(1)  == 1
# digits(4)  == 0
# digits(235) == 15
sub digits {
    my($n) = @_;
    my $product = 1;
    while ($n) {
        my $digit = $n % 10;
        $product *= $digit if $digit % 2 != 0;
        $n = int($n / 10);
    }
    return $product;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&digits;
        if(eq_deeply($candidate->(5),5)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(54),5)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(120),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(5014),5)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(98765),315)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(5576543),2625)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(2468),0)) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();